docs/sources/admin/troubleshooting.md

```diff
@@ -1 +1,86 @@
-# For Administrators: Troubleshooting
+This document's goal is to familiarize matching tool administrators with the right places to look to investigate problems that come up in a deployment of the matching tool. 
+
+## Data Flow Overview
+
+To help with general troubleshooting, it helps to gain some familiarity with how data flows through the application and different containers. Below is a step-by-step overview of data flow.
+
+1. The user uploads a tabular dataset through the webapp UI, which is run by the `webapp` container. This data goes through the [upload_file](https://github.com/dssg/matching-tool/blob/master/webapp/webapp/apis/upload.py#L299) endpoint and is immediately saved to a temporary directory on the filesystem that is shared with the `webapp_worker` container. It tells the `webapp_worker` container to validate the file asynchronously and begins polling the asynchronous job for results.
+2. The `webapp_worker` container runs the (validate_async)[https://github.com/dssg/matching-tool/blob/master/webapp/webapp/apis/upload.py#L193] function asynchronously. This saves some metadata about the upload to the `upload_log` Postgres table and runs a variety of validations on the data. It culminates in uploading the validated and transformed file to storage (S3 by default) at the (`RAW_UPLOADS_PATH`)[https://github.com/dssg/matching-tool/blob/master/webapp.env] location, and returns the validation results.
+3. Assuming the file has no validation problems and the user clicks 'confirm upload', the (merge_file)[https://github.com/dssg/matching-tool/blob/master/webapp/webapp/apis/upload.py#L337] endpoint is called. This loads the stored file into the database under the path `raw_{upload_id}` and merges it with the 'master' table for the relevant jurisdiction and event type.
+
+This 'master' table serves as both the source for the results dashboard. Data is sent from here to the matcher, and data from the matcher is synced back into it. The 'merge' consists of adding rows from the new uploaded file that do not have their primary keys (see primaryKey attribute of the relevent (schema file)[https://github.com/dssg/matching-tool/tree/master/webapp/schemas/uploader] present in the master table, and updating data in the master table with any matching rows from the upload. Placeholder matching ids (of format {event_type}_{internal_person_id}) are populated.
```

Comment:
> This 'master' table serves as both the source for the results dashboard.

Is there supposed to be a second part to this?

Reply:
There used to be; the second part of the sentence was describing how it is interacted with. I instead expanded this into what is now the sentence that immediately follows it. But I forgot to remove the word 'both'.